cirq/ops/common_channels_test.py

```diff
@@ -0,0 +1,119 @@
+    d = cirq.asymmetric_depolarize(0.1, 0.2, 0.3)
+    np.testing.assert_almost_equal(cirq.channel(d),
+                                   (np.sqrt(0.4) * np.eye(2),
+                                    np.sqrt(0.1) * X,
+                                    np.sqrt(0.2) * Y,
+                                    np.sqrt(0.3) * Z))
+
+
+def test_asymmetric_depolarizing_channel_repr():
+    cirq.testing.assert_equivalent_repr(
+        cirq.AsymmetricDepolarizingChannel(0.1, 0.2, 0.3))
+
+
+def test_asymmetric_depolarizing_channel_str():
+    assert (str(cirq.asymmetric_depolarize(0.1, 0.2, 0.3))
+            == 'AsymmetricDepolarizingChannel(p_x=0.1,p_y=0.2,p_z=0.3)')
+
+
+def test_asymmetric_depolarizing_channel_eq():
+    et = cirq.testing.EqualsTester()
+    et.add_equality_group(cirq.asymmetric_depolarize(0.0, 0.0, 0.0))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.0, 0.0, 0.1))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.0, 0.1, 0.0))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.1, 0.0, 0.0))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.1, 0.2, 0.3))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.3, 0.4, 0.3))
+    et.add_equality_group(cirq.asymmetric_depolarize(1.0, 0.0, 0.0))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.0, 1.0, 0.0))
+    et.add_equality_group(cirq.asymmetric_depolarize(0.0, 0.0, 1.0))
+
+
+@pytest.mark.parametrize('p_x,p_y,p_z', (
+    (-0.1, 0.0, 0.0),
+    (0.0, -0.1, 0.0),
+    (0.0, 0.0, -0.1),
+    (0.1, -0.1, 0.1)))
+def test_asymmetric_depolarizing_channel_negative_probability(p_x, p_y, p_z):
+    with pytest.raises(ValueError, match='was less than 0'):
+        cirq.asymmetric_depolarize(p_x, p_y, p_z)
+
+
+@pytest.mark.parametrize('p_x,p_y,p_z', (
+    (1.1, 0.0, 0.0),
+    (0.0, 1.1, 0.0),
+    (0.0, 0.0, 1.1),
+    (0.1, 0.9, 0.1)))
+def test_asymmetric_depolarizing_channel_bigly_probability(p_x, p_y, p_z):
+    with pytest.raises(ValueError, match='was greater than 1'):
+        cirq.asymmetric_depolarize(p_x, p_y, p_z)
+
+
+def test_asymmetric_depolarizing_channel_text_diagram():
+    a= cirq.asymmetric_depolarize(0.1, 0.2, 0.3)
+    assert (cirq.circuit_diagram_info(a) == cirq.CircuitDiagramInfo(
+        wire_symbols=('A(0.1,0.2,0.3)',)))
+
+
+def test_depolarizing_channel():
+    d = cirq.depolarize(0.3)
+    np.testing.assert_almost_equal(cirq.channel(d),
+                                   (np.sqrt(0.7) * np.eye(2),
+                                    np.sqrt(0.1) * X,
+                                    np.sqrt(0.1) * Y,
+                                    np.sqrt(0.1) * Z))
+
+
+def test_depolarizing_channel_repr():
+    cirq.testing.assert_equivalent_repr(cirq.DepolarizingChannel(0.3))
+
+
+def test_depolarizing_channel_str():
+    assert str(cirq.depolarize(0.3)) == 'DepolarizingChannel(p=0.3)'
+
+
+def test_depolarizing_channel_eq():
+    et = cirq.testing.EqualsTester()
+    et.add_equality_group(cirq.depolarize(0.0))
```

Comment:
Need a group with more than one member (you can use the `et.make_equa...` method to produce two from a lambda)

Reply:
Thanks kind of annoying

Reply:
Well, *technically* it compares the objects against themselves. But if you completely forgot to implement `__eq__` the fallback `is` check will pass the tests.

This is just a general feature of tests, I think: no matter how many input cases you give, it might as well just be one if you didn't exercise multiple output cases.